pyspark_datasources/opensky.py

```
@@ -0,0 +1,350 @@
+            StructField("vertical_rate", DoubleType()),
+            StructField("sensors", ArrayType(IntegerType())),
+            StructField("baro_altitude", DoubleType()),
+            StructField("squawk", StringType()),
+            StructField("spi", BooleanType()),
+            StructField("category", IntegerType())
+        ])
+
+    def simpleStreamReader(self, schema: StructType) -> OpenSkyStreamReader:
+        return OpenSkyStreamReader(schema, self.options)
+
+spark.dataSource.register(OpenSkyDataSource)
```

Ruff (0.12.2)

`spark` may be undefined, or defined from star imports

(F405)